src/Microsoft.Health.Dicom.Api/Features/Filters/PopulateOriginalRequestFilterAttribute.cs

```diff
@@ -0,0 +1,48 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Api.Extensions;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Context;
+
+namespace Microsoft.Health.Dicom.Api.Features.Filters;
+
+[AttributeUsage(AttributeTargets.Class)]
+public sealed class PopulateOriginalRequestFilterAttribute : ActionFilterAttribute
+{
+    private readonly IDicomRequestContextAccessor _dicomRequestContextAccessor;
+    private readonly bool _isUpdateEnabled;
+
+    public PopulateOriginalRequestFilterAttribute(
+        IDicomRequestContextAccessor dicomRequestContextAccessor,
+        IOptions<FeatureConfiguration> featureConfiguration)
+    {
+        _dicomRequestContextAccessor = EnsureArg.IsNotNull(dicomRequestContextAccessor, nameof(dicomRequestContextAccessor));
+
+        EnsureArg.IsNotNull(featureConfiguration, nameof(featureConfiguration));
+        _isUpdateEnabled = featureConfiguration.Value.EnableUpdate;
+    }
+
+    public async override Task OnActionExecutionAsync(ActionExecutingContext context, ActionExecutionDelegate next)
+    {
+        EnsureArg.IsNotNull(context, nameof(context));
+        IDicomRequestContext dicomRequestContext = _dicomRequestContextAccessor.RequestContext;
+
+        bool isOriginalVersionRequested = context.HttpContext.Request.IsOriginalVersionRequested();
+
+        if (!_isUpdateEnabled && isOriginalVersionRequested)
+            throw new DicomUpdateFeatureDisabledException();
+
+        dicomRequestContext.IsOriginalRequested = isOriginalVersionRequested;
```

Comment:
Should this be on the context or RetrieveResourceRequest? Since this only applies to retrieve

Reply:
Yeah, i thought about that too. Since we had IsTranscodeREquested on the context, I thought we could use the similar pattern. Just trying to reduce bunch of code changes.

Reply:
Lets not pile on more bad code. This is not generic enough to be on RequestContext